src/android/KnoxPluginEnabled.kt

```diff
@@ -1,127 +1,314 @@
 package com.hrs.knox
 
+import android.app.AlarmManager
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Context.ALARM_SERVICE
+import android.content.Context.TELEPHONY_SERVICE
+import android.content.Intent
+import android.os.Build
+import android.telephony.TelephonyManager
+import com.android.volley.Request
+import com.android.volley.Response
+import com.android.volley.toolbox.StringRequest
+import com.android.volley.toolbox.Volley
+import com.hrs.patient.BuildConfig
+import com.samsung.android.knox.custom.CustomDeviceManager
 import org.apache.cordova.CallbackContext
 import org.apache.cordova.CordovaPlugin
 import org.json.JSONArray
+import org.json.JSONException
 import org.json.JSONObject
 import timber.log.Timber
-import com.samsung.android.knox.custom.CustomDeviceManager
-import com.samsung.android.knox.EnterpriseDeviceManager
-import com.hrs.patient.BuildConfig
-import android.telephony.TelephonyManager
-import android.content.Context.TELEPHONY_SERVICE
+
 
 private const val KNOX_ENABLED = true
 private const val ACTION_IS_ENABLED = "isEnabled"
 private const val ACTION_SHUTDOWN = "shutdown"
 private const val ACTION_REBOOT = "reboot"
 private const val ACTION_GET_VERSION_INFO = "getVersionInfo"
+private const val ACTION_GET_IMEI = "getIMEI"
+private const val ACTION_ACCESS_KNOXAPI = "authenticateAPI"
+
 private const val KEY_KNOX_APP_VERSION = "knoxAppVersion"
-private const val ACTION_GET_IMEI = "getIMEI";
+private const val TIME_SPAN_24_HOURS_MS = 4 * 60 * 60 * 1000
+private var imei: String = "" // Device IMEI that would be saved after the permissions prompts and directly pulled from Telephony
```

Comment:
These should be private member variables defined within the KnoxPlugin class - these other values here are only defined above the class to emulate Java's "private static final" syntax since kotlin does not have that.